lib/r10k/forge/module_release.rb

```diff
@@ -65,34 +75,47 @@ def install(target_dir)
         cleanup
       end
 
-      # Download the module release to {#download_path}
+      # Download the module release to {#download_path} and cache to {#tarball_cache_root}
       #
       # @return [void]
       def download
-        logger.debug1 "Downloading #{@forge_release.slug} from #{PuppetForge::Release.conn.url_prefix} to #{@download_path}"
-        @forge_release.download(download_path)
+        if @tarball_cache_path.exist?
+          logger.debug1 "Cached copy of #{@forge_release.slug} tarball found"
```

Comment:
Nit pick, could you change this to be an action like `Using cached copy of tarball for ..`